website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1076 @@ records.apply("WriteToText",
+```
+
+The `@SchemaCreate` annotation can be used to specify a constructor or a static factory method, in which case the 
+setters and zero-argument constructor can be omitted.
+
+```java
+@DefaultSchema(JavaBeanSchema.class)
+public class TransactionBean {
+  @SchemaCreate
+  Public TransactionBean(String bank, double purchaseAmount) { … }
+  public String getBank() { … }
+  public double getPurchaseAmount() { … }
+}
+```
+
+`@SchemaFieldName` and `@SchemaIgnore` can be used to alter the schema inferred, just like with POJO classes.
+
+##### AutoValue
+Java value classes are notoriously difficult to generate correctly. There is a lot of boilerplate you must create in 
+order to properly implement a value class. AutoValue is a popular framework for easily generating such classes by i
+mplementing a simple abstract base class.
+
+Beam can infer a schema from an AutoValue class. For example:
+
+```java
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class TransactionValue {
+  public abstract String getBank(); 
+  public abstract double getPurchaseAmount();
+}
+```
+
+This is all that’s needed to generate a simple AutoValue class, and the above `@DefaultSchema` annotation tells Beam to
+infer a schema from it. This also allows AutoValue elements to be used inside of `PCollection`s.
+
+`@SchemaFieldName` and `@SchemaIgnore` can be used to alter the schema inferred.
+
+### 6.6. Using Schemas {#using-schemas}
+A schema on a `PCollection` enables a rich variety of relational transforms. The fact that each record is composed of
+named fields allows for simple and readable aggregations that reference fields by name, similar to the aggregations in
+a SQL expression. 
+
+#### 6.6.1. Field selection syntax
+The advantage of schemas is that they allow referencing of element fields by name. Beam provides a selection syntax for
+referencing fields, including nested and repeated fields. This syntax is used by all of the schema transforms when 
+referencing the fields they operate on. The syntax can also be used inside of a DoFn to specify which schema fields to
+process.
+
+Addressing fields by name still retains type safety as Beam will check that schemas match at the time the pipeline graph
+is constructed. If a field is specified that does not exist in the schema, the pipeline will fail to launch. In addition,
+if a field is specified with a type that does not match the type of that field in the schema, the pipeline will fail to
+launch.
+
+##### **Top-level fields**
+In order to select a field at the top level of a schema, the name of the field is specified. For example, to select just
+the user ids from a `PCollection` of purchases one would write (using the `Select` transform)
+
+```java
+purchases.apply(Select.fieldNames(“userId”));
+```
+
+##### **Nested fields**
+Individual nested fields can be specified using the dot operator. For example, to select just the postal code from the
+ shipping address one would write
```

Comment:
Might be a good place to address the question of dots in field names. Do we disallow it? Or do we allow it and it just breaks nested field selection?

Reply:
Similarly for `[]{}*`, do we consider all of these special characters?

Reply:
Added (we should add code to validate that these characters aren't used!)